src/main/java/io/github/thebusybiscuit/slimefun4/utils/ExperienceUtils.java

```diff
@@ -0,0 +1,98 @@
+package io.github.thebusybiscuit.slimefun4.utils;
+
+import org.apache.commons.lang.Validate;
+
+/**
+ * Some helper methods for dealing with experience converting.
+ *
+ * @author kingdom84521
+ *
+ * @see io.github.thebusybiscuit.slimefun4.core.guide.SlimefunGuideImplementation
+ * @see <a href="https://minecraft.fandom.com/wiki/Experience"></a>
+ *
+ */
+public final class ExperienceUtils {
+
+    public static class LevelAndProgress {
+        public Integer level;
+        public Float progress;
+
+        public LevelAndProgress(int level, float progress) {
+            this.level = level;
+            this.progress = progress;
+        }
+    }
+
+    /**
+     * Helper method to calculate the level to exact xp
+     *
+     * @param level
+     *            The level to convert
+     *
+     * @return xp
+     */
+
+    public static float convertLevelToFloatExp(int level) {
+        Validate.isTrue(level >= 0, "Cannot process with a negative level number");
+
+        if (level > 31) {
+            return (float) (4.5 * level * level - 162.5 * level + 2220);
+        } else if (level > 16) {
+            return (float) (2.5 * level * level - 40.5 * level + 360);
+        } else {
+            return level * level + 6 * level;
+        }
+    }
+
+    private static float getExpToNextLevel(int currentLevel) {
+        if (currentLevel > 30) {
+            return (float) 9 * currentLevel - 158;
+        } else if (currentLevel > 15) {
+            return (float) 5 * currentLevel - 38;
+        } else {
+            return (float) 2 * currentLevel + 7;
+        }
```

Comment:
not a fan of these methods. Isnt there a better way to do this?
I dont think the calculation is correct to be honest.